src/dotnet/ToolPackage/LockFileMatcher.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+
+namespace Microsoft.DotNet.ToolPackage
+{
+    internal class LockFileMatcher
+    {
+        private static string[] _pathInLockFilePathInArray;
+        private static string[] _entryPointPathInArray;
+
+        internal static bool MatchesFile(string pathInLockFile, string entryPoint)
+        {
+            _pathInLockFilePathInArray = SplitPathByDirectorySeparator(pathInLockFile);
+            _entryPointPathInArray = SplitPathByDirectorySeparator(entryPoint);
+
+            return EntryPointHasFileAtLeast()
+                   && PathInLockFileHasExcatlyToolsTfmRidAsParentDirectories()
+                   && RestMatchsOtherThanPathInLockFileHasToolsTfmRidAsParentDirectories();
+        }
+
+        private static bool RestMatchsOtherThanPathInLockFileHasToolsTfmRidAsParentDirectories()
+        {
+            string[] pathAfterToolsTfmRid = _pathInLockFilePathInArray.Skip(3).ToArray();
+            return !pathAfterToolsTfmRid
+                .Where((directoryOnEveryLevel, i) => directoryOnEveryLevel != _entryPointPathInArray[i])
+                .Any();
+        }
+
+        private static bool PathInLockFileHasExcatlyToolsTfmRidAsParentDirectories()
```

Comment:
This method name has a typo in it. `Excatly` instead of `Exactly`.

Also, it is not really doing what it claims to be doing. It says it will check for ToolsTfmRid and it only checks that it has three directories there and that it starts with tools.

Reply:
TFM and RID is checked by NuGet already, since it is in asset.json. Maybe a good place to add comment

Reply:
If TFM and RID are already being checked, than don't mention it in this method's name? Seems more confusing to do so, then when someone looks at the implementation, than it does not match the method name.